### ISSUE TITLE
Compiling 0.3.2 error

### ISSUE THREAD

**Issue by 917huB** [leetronics/infnoise#26]
Compiling v0.3.2 on oDroid C2 fails where as 0.3.1 compiles fine. Any suggestions would be appreciated. 

```
make -f Makefile.linux
cc -c -o writeentropy.o writeentropy.c -Wall -Wextra -Werror -std=c99 -O3 -fPIC -I Keccak -I /usr/include/libftdi1 -DGIT_VERSION=\"0.3.2\" -DGIT_COMMIT=\"e80ddd78085abf3d06df2e0d8c08fd33dade78eb\" -DGIT_DATE=\"2019-08-12T21:24:02+02:00\" -DLINUX
writeentropy.c: In function ‘readNumberFromFile’:
writeentropy.c:29:29: error: comparison is always true due to limited range of data type [-Werror=type-limits]
     while( (c = getc(file)) != EOF
                             ^~
cc1: all warnings being treated as errors
make: *** [Makefile.linux:21: writeentropy.o] Error 1
```

**Comment by adudek16** [comment]
Following the changes in the pull request #27 does allow for compiling.

**Comment by danb35** [comment]
I can confirm the error (under Ubuntu Server 22.04 LTS on a Raspberry Pi 4) and the fix.

**Comment by manuel-domke** [comment]
merged PR #27, thanks everyone
